lib/tfctl/executor.rb

```diff
@@ -41,13 +41,23 @@ def run(account_name:, config_name:, log:, cmd: nil, argv: [], unbuffered: true)
             # Create the command
             exec = [cmd] + [subcmd] + args
 
-            # Set environment variables for Terraform
-            env = {
+            env = {}
+            # Copy non-overridden TF_* variables from process environment
+            # (see below for overrides)
+            ENV.each_pair do |variable_name, value|
+                if variable_name.start_with?("TF_")
+                    env[variable_name] = value
+                end
+            end
+
+            # Enforce values of these Terraform environment variables
+            {
                 'TF_INPUT'           => '0',
                 'CHECKPOINT_DISABLE' => '1',
                 'TF_IN_AUTOMATION'   => 'true',
-                # 'TF_LOG'             => 'TRACE'
-            }
+            }.each_pair do |variable_name, value|
```

Comment:
I would suggest to store this essential variables in something like `essentials` variable and then we could do:

```ruby
essentials.merge!(env)
```

looks a bit more readable and idiomatic to me. Overall nicely done!

Reply:
Revised based on feedback